src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -25,6 +25,11 @@ namespace Microsoft.Identity.Client
 {
     internal sealed class ApplicationConfiguration : IAppConfig
     {
+        public ApplicationConfiguration(bool isConfidentialClient) 

```

Comment:
Could't find a better way to do this since ApplicationConfiguration is mutable. We can add logic to property setters but it gets ugly.

Reply:
Alternative can be to have set as internal.

Reply:
I think having this be set in constructor is good, since it forces you to set this property when config is created, so it's not missed by accident somewhere and set to false by default.